### DISCUSSION TITLE
Add an option for circular horizontal and circular vertical modes

### DISCUSSION THREAD

**Discussion by alexvoda** [YaLTeR/niri#223] [Ideas]
Currently scrolling horizontally between tiled windows behaves linearly with a left most en and a right most end.
Similarly scrolling vertically between workspace behaves linearly with a top most and a bottom most workspace.

This request is for implementing 2 options for circular horizontal and circular vertical modes. 

The analogy to use is a circular linked list compared to a linear linked list. After reaching one end, scrolling further would just continue show the elements from the other end. Note: this is not the same as jumping to the beginning of the list after reaching the end.

This would result in enabling infinite scrolling and therefore the ability to fully navigate using just half of the direction keys.

This is not an entirely new concept. The desktop cube as implemented by some desktops is in effect a circular list of virtual desktops.

For a physical analogy, when only horizontal circular mode is enabled, one can think of a stack of cylinders of different diameters (depending on the number of tiled windows) with the windows on the curved surfaces.

This is similar to #41 but applied to both directions.

**Comment by YaLTeR** [comment]
So vertically you still always have one empty workspace, just afterwards it wraps around to the first one?

Overall this is a cool idea but as I said I'm the other discussion, I'm not sure it's entirely in scope, so whether to include it depends on whether it can be implemented in a non disruptive enough way. For workspaces it should actually be quite easy since you never see more than two on screen at once (for now at least, until there's some kind of overview).
